MS visual studio (2008/2010) compatibility

brandonc opened the pull request:
This branch is more concise and has the following improvements
- renamed dll export define to "DLLEXPORT" instead of "WIN32EXPORT"
- includes build configurations for both static and dll libraries
- include paths are now relative
- fewer platform specific #if conditions

Thanks, Vicent

brandonc committed:
visual studio 2010 / win32 port

brandonc committed:
rename solution/project files

brandonc committed:
win32 build

brandonc committed:
forgot some files

brandonc committed:
vs2008 support for upskirt.exe

brandonc committed:
added win32 export for bufnew and bufrelease

brandonc committed:
fix gcc warning when missing WIN32EXPORT define

brandonc committed:
fix gcc warning when missing WIN32EXPORT define

brandonc committed:
MS visual studio (2008/2010) compatibility

brandonc committed:
Add newline to .gitignore, move strings.h include

brandonc committed:
merged

brandonc committed:
trying to merge

brandonc committed:
merged with ms-visual-studio branch

brandonc committed:
merged with upstream

vmg commented:
Wow, this is much more than what I has hoping. :)

Amazing job, I'm going to merge this manually since there are a couple of conflicts to resolve.

GerHobbelt commented:
In case you're interested: merged the Windows port with latest master at   

  git://github.com/GerHobbelt/upskirt.git

plus a few minor tweaks; now includes the smartypants binary as well, plus a few uber-minor code fixes in upskirt itself (a header protection #define typo fix, a size_t which should be treated as an int (negative indexing into the data array there :-) ) and smartypants saying it can't open the binary while it should report the file it failed to access instead.

The whole shebang has seen some basic testing now on both Windows (MSVC2008SP1 / MSVC2010SP1) and Ubuntu/64.

Commit:
Rename Upskirt to `Sundown`

It is by the goodness of God that in our country we have those
three unspeakably precious things:

    freedom of speech,
    freedom of conscience,
    and the prudence never to practice either of them.

                -- Mark Twain, Following the Equator (1897)

Commit:
merged upstream

Commit:
rename project to sundown

brandonc committed:
remove spurious WIN32EXPORT

brandonc commented:
You don't want this history. I merged the master into the pull request branch with all my intermediate changes to it. I'll take a look at @GerHobbelt's fork and maybe send you another pull request.
